src/libraries/System.Text.Json/tests/Common/UnsupportedTypesTests.cs

```diff
@@ -0,0 +1,237 @@
+                    Assert.DoesNotContain("$.Prop", exAsStr);
+                }
+
+                if (type.IsValueType)
+                {
+                    // NSE is not thrown because the serializer handles null.
+                    if (typeof(T).IsValueType)
+                    {
+                        bool nullableOfTUsed = typeof(T) != type;
+                        if (nullableOfTUsed)
+                        {
+                            // Null is handled by the serializer and doesn't throw.
+                            string serialized = await JsonSerializerWrapperForString.SerializeWrapper<T>((T)(object)null);
+                            Assert.Equal("null", serialized);
+
+                            obj.Prop = (T)(object)null;
+                            serialized = await JsonSerializerWrapperForString.SerializeWrapper(obj);
+                            Assert.Equal(@"{""Prop"":null}", serialized);
+
+                            serialized = await JsonSerializerWrapperForString.SerializeWrapper(obj, new JsonSerializerOptions { IgnoreNullValues = true });
+                            Assert.Equal(@"{}", serialized);
+                        }
+                    }
+                }
+                else
+                {
+                    string serialized = await JsonSerializerWrapperForString.SerializeWrapper((T)(object)null);
+                    Assert.Equal("null", serialized);
+
+                    obj.Prop = (T)(object)null;
+                    serialized = await JsonSerializerWrapperForString.SerializeWrapper(obj);
+                    Assert.Equal(@"{""Prop"":null}", serialized);
+
+                    serialized = await JsonSerializerWrapperForString.SerializeWrapper(obj, new JsonSerializerOptions { IgnoreNullValues = true });
+                    Assert.Equal(@"{}", serialized);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task RuntimeConverterIsSupported()
+        {
+            const string Json = "{\"MyIntPtr\":42}";
+            string serialized;
+            JsonSerializerOptions options = new();
+            options.Converters.Add(new IntPtrConverter());
+
+            serialized = await JsonSerializerWrapperForString.SerializeWrapper(new IntPtr(42), options);
+            Assert.Equal("42", serialized);
+
+            IntPtr intPtr = await JsonSerializerWrapperForString.DeserializeWrapper<IntPtr>("42", options);
+            Assert.Equal(42, intPtr.ToInt32());
+
+            ClassWithIntPtr obj = new() { MyIntPtr = new IntPtr(42) };
+            serialized = await JsonSerializerWrapperForString.SerializeWrapper(obj, options);
+            Assert.Equal(Json, serialized);
+
+            obj = await JsonSerializerWrapperForString.DeserializeWrapper<ClassWithIntPtr>(Json, options);
+            Assert.Equal(42, obj.MyIntPtr.ToInt32());
+        }
+
+        [Fact]
+        public async Task CompileTimeConverterIsSupported()
```

Comment:
Would it be possible to also cover the IAsyncEnumerable case in this test? Ditto for runtime converter case.

Reply:
I'll add a test to ensure the converter gets called and doesn't throw NSE, but I think it's outside the scope here to verify actual IAsyncEnumerable semantics of serializer\deserialize; not sure if you wanted that as well.